Readability.js

```diff
@@ -131,6 +131,7 @@ Readability.prototype = {
     prevLink: /(prev|earl|old|new|<|«)/i,
     whitespace: /^\s*$/,
     hasContent: /\S$/,
+    srcsetUrl: /(\S+)(\s+[\d.]+[xw])?,?/g,
```

Comment:
This regular expression and the use with `matchAll` is a bit confusing. Currently, that function is invoked even if the second group doesn't match (due to the `?`) and so it'll just match groups of non-whitespace characters, potentially with no separators at all (because the comma, too, has a `?` behind it).

Can you elaborate a bit on how this is supposed to work? It would seem like it'd be simpler to split the entire string by commas, and whitespace, ie run `str.split(/\s*,\s*/)` and then iterate over the result, then `join` it back together again.

Note also that `matchAll` was not available on `node` prior to 12.4, and the minimum node specified in the package.json is 7; if you want to keep using a match/replace pattern you could probably use the [function callback for `replace` instead](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter).

Reply:
>  It would seem like it'd be simpler to split the entire string by commas and whitespace

Unfortunately it can't works thanks to how Kinja sets their image URLs, for example :

```
https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_180,q_80,w_320/gpf2pi3edjf1zumpctbg.jpg 320w,
https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_264,q_80,w_470/gpf2pi3edjf1zumpctbg.jpg 470w,
https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_80,q_80,w_80/gpf2pi3edjf1zumpctbg.jpg 80w,
https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_180,q_80,w_320/gpf2pi3edjf1zumpctbg.jpg
```

> Can you elaborate a bit on how this is supposed to work?

According to documentation, `srcset` is combined of several strings that joined with comma, where each string is composed by :

1. URL to image; and
2. optionally whitespace followed by size descriptor, which might be width (`w` unit) or density (`x` unit).

With that in mind, I created the regex `(\S+)(\s+[\d.]+[xw])?,?` :

- First group `(\S+)` is used to capture URL. This should works because URL shouldn't contain whitespaces.
- The second group `(\s+[\d.]+[xw])?` is used to capture optional size descriptor.
- The last string in `srcset` will not have comma, that's why the comma in my regex is optional.

If we apply the regex to above srcset examples, we will get results like this :

| First group                                                                                                                       | Second group |
|-----------------------------------------------------------------------------------------------------------------------------------|--------------|
| https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_180,q_80,w_320/gpf2pi3edjf1zumpctbg.jpg | 320w      |
| https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_264,q_80,w_470/gpf2pi3edjf1zumpctbg.jpg | 470w      |
| https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_80,q_80,w_80/gpf2pi3edjf1zumpctbg.jpg   | 80w       |
| https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_180,q_80,w_320/gpf2pi3edjf1zumpctbg.jpg | `null`       |

Now we just need to convert first group to absolute URL, the combine it with second group.

Reply:
> Note also that `matchAll` was not available on node prior to 12.4

It has been fixed in [`ffdca4b`](https://github.com/mozilla/readability/pull/590/commits/ffdca4b50557b0b1c841310bf2843efc6498181c).